pkg/reconciler/source/duck/duck_test.go

```diff
@@ -250,6 +252,50 @@ func TestAllCases(t *testing.T) {
 			WantCreates: []runtime.Object{
 				makeEventType("my-type-1", "http://my-source-1"),
 			},
+		}, {
+			Name: "no op",
+			Objects: []runtime.Object{
+				makeSource([]duckv1.CloudEventAttributes{{
+					Type:   "my-type-1",
+					Source: "http://my-source-1",
+				}}),
+				func() runtime.Object {
+					s := makeSourceCRD([]eventTypeEntry{{
+						Type:        "my-type-1",
+						Schema:      "/some-schema-from-crd",
+						Description: "This came from the annotation in a crd for the source.",
+					}})
+					s.Annotations[eventing.EventTypesAnnotationKey] = "something that is not valid json"
+					return s
+				}(),
+				makeEventType("my-type-1", "http://my-source-1"),
+			},
+			Key: testNS + "/" + sourceName,
+			WantCreates: []runtime.Object{},
+		}, {
+			Name: "no op with namespace",
+			Objects: []runtime.Object{
+				makeSource([]duckv1.CloudEventAttributes{{
+					Type:   "my-type-1",
+					Source: "http://my-source-1",
+				}}),
+				func() runtime.Object {
+					s := makeSourceCRD([]eventTypeEntry{{
+						Type:        "my-type-1",
+						Schema:      "/some-schema-from-crd",
+						Description: "This came from the annotation in a crd for the source.",
+					}})
+					s.Annotations[eventing.EventTypesAnnotationKey] = "something that is not valid json"
+					return s
+				}(),
+				func() runtime.Object {
+					et := makeEventType("my-type-1", "http://my-source-1")
+					et.Spec.Reference.Namespace = testNS
+					return et
+				}(),
+			},
+			Key: testNS + "/" + sourceName,
+			WantCreates: []runtime.Object{},
```

Comment:
nit (for the linter):
```suggestion
			Key:         testNS + "/" + sourceName,
			WantCreates: []runtime.Object{},
```